[14.0][ADD] cetmix_tower_tldextract: new module

Mohajiro opened the pull request:
Created a new module for 'tldextract' library.

The 'cetmix_tower_ovh' module has been updated to depend on this new module.

Task: 4737

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a new module that safely wraps the Python tldextract library for domain parsing within Cetmix Tower commands.
  - Added a demo command showcasing domain extraction capabilities.
  - Provided help text and usage documentation for the new domain parsing functionality.

- **Documentation**
  - Added comprehensive README, usage instructions, and HTML documentation for the new module.
  - Updated license badge spelling in existing documentation for consistency.

- **Chores**
  - Updated module dependencies to improve modularity and reduce unnecessary package loading.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Mohajiro committed:
[ADD] cetmix_tower_tldextract: new module

Created a new module for 'tldextract' library.

The 'cetmix_tower_ovh' module has been updated to depend on this new
module.

Task: 4737

CetmixGitDrone commented:
Hi @GSLabIt,
some modules you are maintaining are being modified, check this out!

Mohajiro committed:
[FIX] cetmix_tower_ovh: manifest

Fix of dependencies in cetmix_tower_ovh

Task: 4737

ivs-cetmix commented on cetmix_tower_tldextract/i18n/cetmix_tower_ovh.pot:
This file will be generated automatically. Please remove it.

Mohajiro replied:
Done

ivs-cetmix commented on cetmix_tower_tldextract/__manifest__.py:
```diff
@@ -0,0 +1,18 @@
+    "version": "14.0.1.0.0",
+    "category": "Productivity",
+    "license": "AGPL-3",
+    "author": "Cetmix",
```
Keep all the [authors](https://github.com/cetmix/cetmix-tower/blob/19564e03dce457f3ddbf40fa809cf24968ab90db/cetmix_tower_ovh/__manifest__.py#L11C24-L11C38) and [maintainers](https://github.com/cetmix/cetmix-tower/blob/19564e03dce457f3ddbf40fa809cf24968ab90db/cetmix_tower_ovh/__manifest__.py#L12) of the original module.

ivs-cetmix reviewed (changes requested):
You need to adjust the [helper function texts](https://github.com/cetmix/cetmix-tower/blob/14.0/cetmix_tower_ovh/models/constants.py) as well.
Same with the [demo data](https://github.com/cetmix/cetmix-tower/blob/14.0/cetmix_tower_ovh/demo/demo_data.xml).

Mohajiro committed:
[IMP] cetmix_tower_tldextract: Ivan's proposals

Add the corrects authors and maintainers.
Add demo data and helper function texts.
Delete cetmix_tower_ovh.pot.

Task: 4737

ivs-cetmix commented on cetmix_tower_ovh/models/cx_tower_command.py:
```diff
@@ -4,11 +4,13 @@
 # Wrap ovh safely
 ovh = wrap_module(__import__("ovh"), ["Client"])
-tldextract = wrap_module(__import__("tldextract"), ["extract"])
+tldextract = tldextract_wrapper.tldextract
```
Why is it here?  It must be moved out of the module completely.

ivs-cetmix commented on cetmix_tower_ovh/models/cx_tower_command.py:
```diff
@@ -4,11 +4,13 @@
 from odoo import api, models
 from odoo.tools.safe_eval import wrap_module
 
+from odoo.addons.cetmix_tower_tldextract.models import tldextract_wrapper
```
?

ivs-cetmix reviewed (changes requested):
The task was updated. Please check it and update the code accordingly.
